build/jemalloc.m4

```diff
@@ -58,7 +58,7 @@ if test "$has_jemalloc" != "no"; then
   if test "$jemalloc_base_dir" != "/usr"; then
     TS_ADDTO(CPPFLAGS, [-I${jemalloc_include}])
     TS_ADDTO(LDFLAGS, [-L${jemalloc_ldflags}])
-    TS_ADDTO(LDFLAGS, [-Wl,--add-needed -L${jemalloc_ldflags} -Wl,-rpath,${jemalloc_ldflags} -Wl,--no-as-needed])
```

Comment:
@duke8253 is this typo of `--as-needed` option? Because this line has `--no-as-needed` option at the end and it restores the default behavior.

If we really need `--add-needed` option here, `--copy-dt-needed-entries` is the replacement, but `lld` doesn't have it either.

Reply:
I don't think it's a typo because I remember coming back to this and thought it was a typo, but changing it broke stuff. Though I can't remember exactly what was it now, vaguely recall that something about the order of specifying those options made a difference.